From 230f6e68e2e08f2cfd65771e32d9a05b6a8b4eca Mon Sep 17 00:00:00 2001
From: Lenny Szubowicz <lszubowi@redhat.com>
Date: Fri, 13 Jun 2014 18:56:41 -0400
Subject: [acpi] acpica: Add "Windows 2013" string to _OSI support

Message-id: <1402685801-10871-1-git-send-email-lszubowi@redhat.com>
Patchwork-id: 83933
O-Subject: [RHEL 7.1 PATCH BZ 1087674] ACPICA: Add "Windows 2013" string to _OSI support.
Bugzilla: 1087674
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1087674

Description:

    Direct application of upstream patch to facilitate ACPI enumeration
    with some Broadwell BIOS's.

Upstream Status: faae404ebdc6bba744919d82e64c16448eb24a36

Brew: http://brewweb.devel.redhat.com/brew/taskinfo?taskID=7577958

Tested:

    Test booted on a Intel Broadwell-based proto.
    - AML code on this platform inquired about Win 2013
      and invoked this changed code.
    Sanity booted on an Intel Haswell-based system.

commit faae404ebdc6bba744919d82e64c16448eb24a36
Author: Bob Moore <Robert.Moore@intel.com>
Date:   Tue Feb 11 10:25:27 2014 +0800

    ACPICA: Add "Windows 2013" string to _OSI support.

    This urgent patch is cherry picked from ACPICA upstream.
    It is reported that some platforms fail to boot without this new _OSI
    string.

    This change adds this string for Windows 8.1 and Server 2012 R2.

    Reported-by: Zhang Rui <rui.zhang@intel.com>
    Signed-off-by: Bob Moore <Robert.Moore@intel.com>
    Signed-off-by: Lv Zheng <lv.zheng@intel.com>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/acpi/acpica/utosi.c b/drivers/acpi/acpica/utosi.c
index 7e80772..891c304 100644
--- a/drivers/acpi/acpica/utosi.c
+++ b/drivers/acpi/acpica/utosi.c
@@ -74,6 +74,7 @@ static struct acpi_interface_info acpi_default_supported_interfaces[] = {
  {"Windows 2006 SP2", NULL, 0, ACPI_OSI_WIN_VISTA_SP2}, /* Windows Vista SP2 - Added 09/2010 */
  {"Windows 2009", NULL, 0, ACPI_OSI_WIN_7}, /* Windows 7 and Server 2008 R2 - Added 09/2009 */
  {"Windows 2012", NULL, 0, ACPI_OSI_WIN_8}, /* Windows 8 and Server 2012 - Added 08/2012 */
+ {"Windows 2013", NULL, 0, ACPI_OSI_WIN_8}, /* Windows 8.1 and Server 2012 R2 - Added 01/2014 */
 
  /* Feature Group Strings */
 
-- 
1.7.1